kube/src/api/crds.rs

```diff
@@ -55,12 +60,16 @@ impl CrBuilder {
     /// Create a CrBuilder specifying the CustomResource's kind
     ///
     /// The kind must not be plural and it must be in PascalCase
-    fn kind(kind: &str) -> Self {
-        assert!(to_plural(kind) != kind); // no plural in kind
-        assert!(is_pascal_case(&kind)); // PascalCase kind
-        Self {
-            kind: kind.into(),
-            ..Default::default()
+    fn try_kind(kind: &str) -> Result<Self, &'static str> {
+        if to_plural(kind) == kind {
```

Comment:
What is more idiomatic? If/else or if with early return?

Reply:
`if`/`else` is fine here IMO.